code/game/objects/items/weapons/RCD.dm

```diff
@@ -131,6 +131,61 @@
 	GLOB.rcd_list -= src
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "<span class='notice'>Deconstructing user...</span>")
+		var/obj/effect/temp_visual/rcd_effect/reverse/suicide_A = new(suicide_tile)
+		if(!do_after(user, 5 SECONDS))					// Deconstruction of most structures takes 5 seconds, wait for the animation to finish.
+			QDEL_NULL(suicide_A)
+			return
+		if(!(user.l_hand == src) && !(user.r_hand == src))	// Do not commit die if the RCD isn't in your hands.
+			return SHAME								// If this triggers, someone probably chopped off your hand mid-suicide for some reason.
+		useResource(5, user)	// Consume ammo.
+		user.visible_message("<span class='suicide'>[user] deconstructs [user.p_they()]self with [src]!</span>")
```

Comment:
p_they() is the one that should be used here? Tested the PR today and i think i saw something odd with this message

Reply:
changed to p_themselves()